code/RecaptchaField.php

```diff
@@ -155,79 +90,55 @@ public function Field($properties=array()) {
 		}
 
 		$html = '';
-		
-		// Add javascript options as a <script> tag preceeding the JS that actually includes the
-		// recaptcha
-		if(!empty($this->jsOptions)) {
-			$html .= "<script type=\"text/javascript\">//<![CDATA[\n"
-				. "var RecaptchaOptions = " . $this->getJsOptionsString()
-				. "//]]></script>";
-		}
 
 		$previousError = Session::get("FormField.{$this->form->FormName()}.{$this->getName()}.error");
 		Session::clear("FormField.{$this->form->FormName()}.{$this->getName()}.error");
 
 		// iframe (fallback)
-		$iframeURL = ($this->useSSL) ? 'https://' : 'http://';
-		$iframeURL .= sprintf(self::$recaptcha_noscript_url, self::$public_api_key);
+		$iframeURL = sprintf(self::$recaptcha_noscript_url, self::$public_api_key);
 		if(!empty($previousError)) $iframeURL .= "&error={$previousError}";
 		
 		// js (main logic)
-		$jsURL = ($this->useSSL) ? 'https://' : 'http://';
-		$jsURL .= sprintf(self::$recaptcha_js_url, self::$public_api_key);
+		$jsURL = sprintf(self::$recaptcha_js_url, self::$public_api_key);
 		if(!empty($previousError)) $jsURL .= "&error={$previousError}";
-	
-		
-		if($this->useAjaxAPI) {
-			$ajaxURL = ($this->useSSL) ? 'https://' : 'http://';
-			$ajaxURL .= self::$recaptcha_ajax_url;
-			Requirements::javascript($ajaxURL);
-			Requirements::customScript('
-					//<![CDATA[
-					Recaptcha.create("' . self::$public_api_key . '",
-					"' . $this->getName() . '", {
-					   callback: Recaptcha.focus_response_field
-					});
-				//]]>
-			');
-		} else {
-			if(!empty($this->jsOptions) && isset($this->jsOptions['theme']) && $this->jsOptions['theme'] === 'custom'){
-				$html .= $this->renderWith("CustomRecaptchaField");
-			}
-			$html .= '
-				<script type="text/javascript" src="' . $jsURL . '">
-				</script>
-			';
+
+		// turn options array into data attributes
+		$option_string = '';
+		foreach($this->options as $option => $value) {
+			$option_string .= ' data-' . htmlentities($option) . '="' . htmlentities($value) . '"';
 		}
+	
+		Requirements::javascript($jsURL);
+		$html .= '<div class="g-recaptcha" id="' . $this->getName() . '"
+			data-sitekey="' . self::$public_api_key . '"' . $option_string . '></div>';
 		
 		// noscript fallback
-		$html .= '<noscript>
-			<iframe src="' .$iframeURL . '" height="300" width="500"></iframe>
-			<br />
-			<textarea name="recaptcha_challenge_field" rows="3" cols="40"></textarea>
-			<input type="hidden" name="recaptcha_response_field" value="manual_challenge" />
-			</noscript>';
-		return $html;
-	}
-	
-	/**
-	 * Transform options from PHP-array to javascript-object encapsulated
-	 * in a string.
-	 * 
-	 * @todo switch to json_encode once we move requirements to PHP 5.2+
-	 * @return string
-	 */
-	public function getJsOptionsString() {
-		$js = "{";
-		$i=1;
-		foreach($this->jsOptions as $k => $v) {
-			$js .= "'{$k}':'{$v}'";
-			if($i < count($this->jsOptions)) $js .= ',';
-			$i++;
-		}
-		$js .= "}";
+		$html .= <<<EOF
```

Comment:
@HAZARDU5: It works fine as-is (for me at least), just the tests aren't checking the right things. As I said I'll discuss this with @chillu once he's back in the office, but in the meantime I'm using it on a project.